targets/TARGET_NORDIC/TARGET_NRF5/sleep.c

```diff
@@ -75,6 +77,15 @@ void hal_sleep(void)
 
 void hal_deepsleep(void)
 {
+    if (us_ticker_is_enabled()) {
+        /* Disable timer used for us ticker. */
+        nrf_timer_task_trigger(NRF_TIMER1, NRF_TIMER_TASK_STOP);
+    }
+
     hal_sleep();
     //   NRF_POWER->SYSTEMOFF=1;
+
+    if (us_ticker_is_enabled()) {
+        nrf_timer_task_trigger(NRF_TIMER1, NRF_TIMER_TASK_START);
```

Comment:
Maybe a dedicated function can takes care of that and update `us_ticker_enabled` at the same time.

Reply:
`us_ticker_enabled` variable informs if ticker has been initialised (enabled). Enabled ticker can not be disabled, because there is no API to disable the ticker. 
I assume that during the deep-sleep ticker is still enabled, but its counter is frozen. In case when ticker is enabled on wake-up we need to revive its counter.

Reply:
I understood the logic before making my comments however having a hidden undocumented function named `us_ticker_is_enabled` returning true when the us_ticker is at best misleading.

If you want to keep the logic, please rename the function `us_ticker_is_enabled` into something less ambiguous. `us_ticker_is_initialized` may be a relevant name . 

Note that soon there will be API to disable tickers: https://github.com/ARMmbed/mbed-os/pull/5856 .